utils/s3/client.cc

```diff
@@ -223,6 +232,64 @@ future<client::stats> client::get_object_stats(sstring object_name) {
     co_return st;
 }
 
+future<client::TagSet> client::get_object_tagging(sstring object_name) {
+    // see https://docs.aws.amazon.com/AmazonS3/latest/API/API_GetObjectTagging.html
+    auto req = http::request::make("GET", _host, fmt::format("{}?tagging", object_name));
+    s3l.trace("GET {} tagging", object_name);
+    TagSet tagging;
+    co_await make_request(std::move(req),
+                          [&tagging] (const http::reply& reply, input_stream<char>&& in) mutable -> future<> {
+        auto& tagset = tagging;
+        auto input = std::move(in);
+        auto body = co_await util::read_entire_stream_contiguous(input);
+        auto doc = std::make_unique<rapidxml::xml_document<>>();
+        try {
+            doc->parse<0>(body.data());
+        } catch (const rapidxml::parse_error& e) {
+            s3l.warn("cannnot parse tagging response: {}", e.what());
+            throw std::runtime_error("cannot parse tagging response");
+        }
+        auto tagset_node = doc->first_node("Tagging")->first_node("TagSet");
+        for (auto tag_node = tagset_node->first_node("Tag"); tag_node; tag_node = tag_node->next_sibling()) {
```

Comment:
What happens here if there are zero tags? Above I assumed that first_node() throws if there is no such tag (otherwise, the code above could cause a null pointer dereference!) but here it seems it is a null pointer. So what's going on?

Reply:
it returns a `nullptr`. the newly added test actually verify this behavior.